msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/msalonly/basic/TestCase2517381.java

```diff
@@ -0,0 +1,163 @@
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+
+        // install old MsalTestApp then acquires token interactively and silently
+        MsalTestApp msalTestApp = new MsalTestApp();
+        msalTestApp.uninstall();
+        msalTestApp.installOldApk();
+        msalTestApp.launch();
+        msalTestApp.handleFirstRun();
+
+        final MicrosoftStsPromptHandlerParameters promptHandlerParametersMsal = MicrosoftStsPromptHandlerParameters.builder()
+                .prompt(PromptParameter.SELECT_ACCOUNT)
+                .loginHint(username)
+                .sessionExpected(false)
+                .broker(mBroker)
+                .expectingBrokerAccountChooserActivity(false)
+                .expectingProvidedAccountInBroker(false)
+                .expectingLoginPageAccountPicker(false)
+                .expectingProvidedAccountInCookie(false)
+                .consentPageExpected(false)
+                .passwordPageExpected(true)
+                .speedBumpExpected(false)
+                .registerPageExpected(false)
+                .enrollPageExpected(false)
+                .staySignedInPageExpected(false)
+                .verifyYourIdentityPageExpected(false)
+                .howWouldYouLikeToSignInExpected(false)
+                .build();
+
+        String token = msalTestApp.acquireToken(username, password, promptHandlerParametersMsal,true);
+        Assert.assertNotNull(token);
+
+        // then acquire token silently and validate the token
+        msalTestApp.handleBackButton();
+        String silentToken = msalTestApp.acquireTokenSilent();
+        Assert.assertNotNull(silentToken);
+
+        // install old OneAuthTestApp then acquires token interactively and silently
+        final OneAuthTestApp oneAuthApp = new OneAuthTestApp();
+        oneAuthApp.installOldApk();
+        oneAuthApp.launch();
+        oneAuthApp.handleFirstRun();
+
+        final FirstPartyAppPromptHandlerParameters promptHandlerParametersOneAuth = FirstPartyAppPromptHandlerParameters.builder()
+                .broker(mBroker)
+                .prompt(PromptParameter.LOGIN)
+                .loginHint(username)
+                .consentPageExpected(false)
+                .speedBumpExpected(false)
+                .sessionExpected(false)
+                .expectingBrokerAccountChooserActivity(false)
+                .expectingLoginPageAccountPicker(false)
+                .enrollPageExpected(false)
+                .build();
+        oneAuthApp.addFirstAccount(username, password, promptHandlerParametersOneAuth);
+        oneAuthApp.confirmAccount(username);
+
+        // Hit back button to go on launch screen
+        oneAuthApp.handleBackButton();
+
+        final String silentTokenOneAuth = oneAuthApp.acquireTokenSilent();
+        Assert.assertFalse(TextUtils.isEmpty(silentTokenOneAuth));
+        oneAuthApp.assertSuccess();
```

Comment:
what does this assertSuccess here mean? Would the above line at 114 not be sufficent?

Reply:
114 is to check if the UI is empty. 115 is to make sure UI contains "Success", account id and result area is not empty to make sure the token is valid.